blob_io.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2022 Yasuhiro Matsumoto <mattn.jp@gmail.com>.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+package sqlite3
+
+/*
+#ifndef USE_LIBSQLITE3
+#include "sqlite3-binding.h"
+#else
+#include <sqlite3.h>
+#endif
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"io"
+	"runtime"
+	"unsafe"
+)
+
+// SQLiteBlob implements the SQLite Blob I/O interface.
+type SQLiteBlob struct {
+	conn *SQLiteConn
+	blob *C.sqlite3_blob
+	size int
+	offs int
+}
+
+// Blob opens a blob.
```

Comment:
Please fill this out with more details. Linking to the relevant SQLite3 docs will also be helpful.